specification/refarch.adoc

```diff
@@ -3,142 +3,184 @@
+For TVM accesses for instruction fetch and page walks (virtual address
+translation), isolated/confidential memory is required to enforce the
+following security properties:
+
+* TEE Instruction fetch - security property: TVM/TSM must not fetch code
+from untrusted/shared memory (this property must be enforced by the hart e.g. MMU)
+* TEE Paging structure walk - security property: TVM/TSM must not locate
+page tables in untrusted shared memory (for TVM, G-stage walks enforced
 to be in confidential memory by the TSM)
-* TEE data fetch - security property: TVM/TSM allowed to relax data 
-accesses to non-confidential memory (via MTT) to allow for IO accesses.
+* TEE data fetch - security property: The TVM via the TSM may be allowed to
+relax data accesses to non-confidential memory (via MTT) to allow for IO
+accesses.
 
 ==== Cached translations/TLB management
-During confidential memory conversion or reclamation, the TCB must 
-enforce that stale translations cached in harts are not accessible 
+During confidential memory conversion or reclamation, the TCB must
+enforce that stale translations cached in harts are not accessible
 to the untrusted host or another TVM context.
-During confidential memory assignment to a TVM (or during conversion 
-of confidential memory to shared), the TCB must enforce that stale 
-translations may not be held to memory yielded by a TVM (and used 
-by the host for another TVM or VM or the host). 
-These properties are implemented by the TSM in conjunction with 
-the HW TCB via the proposed TEEI.
+During confidential memory assignment to a TVM (or during conversion
+of confidential memory to shared), the TCB must enforce that stale
+translations may not be held to memory yielded by a TVM (and used
+by the host for another TVM or VM or the host).
+These properties are implemented by the TSM in conjunction with
+the HW (e.g. MTT cache invalidations) via the proposed COVH interface.
 
 === TSM initialization
 
-The CoVE architecture requires a hardware Root-of-trust for supporting 
-TCB measurement, reporting and storage <<R8>>. The Root-of-trust for 
-Measurement (RTM) is defined as the TCB component that performs a 
-measurement of an entity and protects it for subsequent reporting. The 
-Root-of-trust for Reporting (RTR) is typically a HW RoT that reliably 
-provides authenticity and non-repudiation services for the purposes of 
-attesting to the origin, integrity and security version of platform TCB 
-components. Each TCB layer should have associated security version numbers 
-(SVN) to allow for TCB recovery in the event of security vulnerabilities 
+The CoVE architecture requires a hardware Root-of-trust for supporting
+TCB measurement, reporting and storage <<R8>>. The Root-of-trust for
+Measurement (RTM) is defined as the TCB component that performs a
+measurement of an entity and protects it for subsequent reporting. The
+Root-of-trust for Reporting (RTR) is typically a HW RoT that reliably
```

Comment:
"protects it" -> should be clarified. Is this refering to protecting the measurement register. A RTM is not responsible for protecting the integrity of the MLE.

Reply:
fixed it (meant "signs it")